In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

print("Notebook is working")


Matplotlib is building the font cache; this may take a moment.


Notebook is working


# Introduction
The purpose of this project is for me to practice using python as an analysis tool. 

## Data Inspection
We'll first inspect the raw data that we have obtained. By using `os` module, we'll be able to interact with underlying operating system, such as viewing and accessing directories.

In [61]:
import os
# Ensure working directory is the project root (Netflix_Movies)
PROJECT_ROOT = os.path.abspath("Netflix_Movies")

# If we're already inside Netflix_Movies, this is harmless
if not os.getcwd().endswith("Netflix_Movies"):
    os.chdir(PROJECT_ROOT)

os.getcwd()

'd:\\2. Areas\\Career_Exploration\\Data_Analyst\\Python\\Kaggle_Notebooks\\Netflix_Movies'

We have found the data. Let's read the file and asign its content into the variable `df`.

In [63]:
df = pd.read_csv("Data/netflix_titles.csv.zip")

Let's first check what the data looks like using.

In [ ]:
df.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

Using `df.head()`, we can see what the data is like. It seems that there are only two types of data here, either each row represents a Movie or a TV Show. Some values seem to be missing, indicated by the "NaN"s. 

In [ ]:
df.shape

(8807, 12)

Using `df.shape` we can see the dimension of the file. These numbers tell us that the data has 12 rows with 8,807 obvervations.

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


Using `df.info()` we can see the summary of each column, like their:
- total row numbers
- total column numbers
- data type
- number of missing values

The dataset contains 12 columns and 8,807 observations (rows).
Most columns are stored as `object` types, with `release_year` being the only numeric column (`int64`).

Some columns contain missing values, including `director`, `cast`, `country`, `date_added`, `rating`, and `duration`. These columns will need to be treated before further processing.